streaming/src/main/scala/org/apache/spark/streaming/dstream/PairDStreamFunctions.scala

```diff
@@ -350,6 +349,18 @@ class PairDStreamFunctions[K, V](self: DStream[(K, V)])
     )
   }
 
+  /** TODO: Add scala docs */
+  def trackStateByKey[S: ClassTag, T: ClassTag](
```

Comment:
Actually, `ClassTags` in `TrackStateSpec` and `TrackStateSpecImpl` are unnecessary. We can remove them and keep `S: ClassTag, T: ClassTag` in `trackStateByKey`.